runtime/autoload/dist/ft.vim

```diff
@@ -1298,5 +1298,21 @@ export def FTvba()
   endif
 enddef
 
+export def Detect_UCI_statements(): bool
+  # Match a config or package statement at the start of the line.
+  const config_or_package_statement = '^\s*\(\(c\|config\)\|\(p\|package\)\)\s\+\S'
+  # Match a line that is either all blank or blank followed by a comment
+  const comment_or_blank = '^\s*\(#.*\)\?$'
+
+  # Return true iff the file has a config or package statement near the
+  # top of the file and all preceding lines were comments or blank.
+  return getline(1) =~# config_or_package_statement
+  \   || getline(1) =~# comment_or_blank
+  \      && (   getline(2) =~# config_or_package_statement
+  \          || getline(2) =~# comment_or_blank
+  \             && getline(3) =~# config_or_package_statement
```

Comment:
is this okay to only check the first 3 lines? The logic is a bit ambigious: `expr || expr && expr`.

Reply:
What's ambiguous? && has higher precedence than ||, so the meaning of the code is well defined.

> is this okay to only check the first 3 lines?

I don't understand this question either, sorry. Can you rephrase it, please?

Reply:
> What's ambiguous? && has higher precedence than ||, so the meaning of the code is well defined.

It might be well-defined, but I prefer to be explict, so I don't have to remember it

Reply:
> I don't understand this question either, sorry. Can you rephrase it, please?

Is it enough to only check the first 3 lines? Can't there be 20 line comments first, and only then a `config_or_package_statement`?

Reply:
&& has higher precedence in every programming language I remember using it in, but sure, I don't really care if there are extra parentheses in there or not. Are you requesting that I add them?

The heuristic can be wrong if it checks 0 or 1000 lines. 3 lines was chosen arbitrarily. E.g. it would be better if a blank file with path `sshfs/router1/etc/config/firewall` gets UCI syntax highlighting, but the current detection code would not assign it the uci filetype.

Honestly, I regret mentioning that the path matches perhaps too many files 🥲

Reply:
alright fine.